datalad/support/gitrepo.py

```diff
@@ -2392,14 +2397,26 @@ def push(self, remote=None, refspec=None, all_remotes=False,
         if all_remotes:
             # be nice to the elderly
             all_ = True
-        return list(
-            self.push_(
-                remote=remote,
-                refspec=refspec,
-                all_=all_,
-                git_options=git_options,
+
+        push_refspecs = [refspec]
+        if remote and refspec:
+            refspec = ensure_list(refspec)
+            if not any(b.startswith(remote + "/") for b in self.get_remote_branches()):
```

Comment:
we have already multiple points in the code which beg for `get_remote_branches` to get a kwarg `remote=None` and filter / strip remote name if `remote` is provided.  I could implement/fold that in this PR if deemed desired.

Reply:
FWIW, this condition is no longer in the PR